content/en/docs/concepts/architecture/_index.md

```diff
@@ -5,4 +5,162 @@ description: >
+
+### kube-apiserver
+
+{{< glossary_definition term_id="kube-apiserver" length="all" >}}
+
+### etcd
+
+{{< glossary_definition term_id="etcd" length="all" >}}
+
+### kube-scheduler
+
+{{< glossary_definition term_id="kube-scheduler" length="all" >}}
+
+### kube-controller-manager
+
+{{< glossary_definition term_id="kube-controller-manager" length="all" >}}
+
+There are many different types of controllers. Some examples of them are:
+
+- Node controller: Responsible for noticing and responding when nodes go down.
+- Job controller: Watches for Job objects that represent one-off tasks, then creates Pods to run those tasks to completion.
+- EndpointSlice controller: Populates EndpointSlice objects (to provide a link between Services and Pods).
+- ServiceAccount controller: Create default ServiceAccounts for new namespaces.
+
+The above is not an exhaustive list.
+
+### cloud-controller-manager
+
+{{< glossary_definition term_id="cloud-controller-manager" length="short" >}}
+
+The cloud-controller-manager only runs controllers that are specific to your cloud provider. If you are running Kubernetes on your own premises, or in a learning environment inside your own PC, the cluster does not have a cloud controller manager.
+
+As with the kube-controller-manager, the cloud-controller-manager combines several logically independent control loops into a single binary that you run as a single process. You can scale horizontally (run more than one copy) to improve performance or to help tolerate failures.
+
+The following controllers can have cloud provider dependencies:
+
+- Node controller: For checking the cloud provider to determine if a node has been deleted in the cloud after it stops responding
+- Route controller: For setting up routes in the underlying cloud infrastructure
+- Service controller: For creating, updating and deleting cloud provider load balancers
+
+## Node Components
+
+Node components run on every node, maintaining running pods and providing the Kubernetes runtime environment.
+
+### kubelet
+
+{{< glossary_definition term_id="kubelet" length="all" >}}
+
+### kube-proxy
+
+{{< glossary_definition term_id="kube-proxy" length="all" >}}
+
+### Container runtime
+
+{{< glossary_definition term_id="container-runtime" length="all" >}}
+
+## Addons
+
+Addons use Kubernetes resources ({{< glossary_tooltip term_id="daemonset" >}}, {{< glossary_tooltip term_id="deployment" >}}, etc) to implement cluster features. Because these are providing cluster-level features, namespaced resources for addons belong within the `kube-system` namespace.
+
+Selected addons are described below; for an extended list of available addons, please see [Addons](/docs/concepts/cluster-administration/addons/).
```

Comment:
```suggestion
## Add-ons

Add-ons use Kubernetes resources ({{< glossary_tooltip term_id="daemonset" >}}, {{< glossary_tooltip term_id="deployment" >}}, etc) to implement cluster features. Because these are providing cluster-level features, namespaced resources for add-ons belong within the `kube-system` namespace.

Selected add-ons are described below; for an extended list of available add-ons, please see [Addons](/docs/concepts/cluster-administration/addons/).
```
It's easy to confuse [add-on](https://en.wikipedia.org/wiki/Add-on) with [addon](https://en.wikipedia.org/wiki/Addon).

Reply:
This suggestion is debatable since the file name here is addons, `content/en/docs/concepts/cluster-administration/addons.md`.
I'll be happy to open a PR to fix the file name. What do you think, @sftim?

Reply:
I think this text may have been moved rather than written new. We avoid providing much corrective feedback to the parts of a change where a contributor is only moving text around.

Reply:
Thank you for the feedback @network-charles!

Given that this text was moved from an existing file and to maintain consistency with current documentation, I suggest keeping the term 'addons' without the hyphen. If a broader change to 'add-ons' is desired across the documentation, that might be better addressed in a separate, dedicated PR.

Reply:
Oh, ok. I thought the text was new. Yes, I'll open a new PR addressing the 'addons' issue when this PR has been merged.

Reply:
Sounds like a great idea, @network-charles. I appreciate your attention to detail.